ambry-router/src/test/java/com.github.ambry.router/PutManagerTest.java

```diff
@@ -614,23 +617,25 @@ private CountDownLatch submitPut()
       throws Exception {
     final CountDownLatch doneLatch = new CountDownLatch(requestAndResultsList.size());
     // This check is here for certain tests (like testConcurrentPuts) that require using the same router.
-    if (router == null || !router.isOpen()) {
+    if (instantiateNewRouterForPuts) {
```

Comment:
"things that should not have closed the router but did" - that is a fair point - though that should be tested using a specific test in my opinion. Not via sharing states between different tests that are supposed to be independent.

What the tests now check is that when the router is closed at the end of each test (explicitly or otherwise), regardless of the context the router might have gone through for that test, the cleanup happens properly.

In general, I prefer instantiating things cleanly every time so that states do not get shared.